### ISSUE TITLE
Unnecessary garbage creation in PlayGamesHelperObject

### ISSUE THREAD

**Issue by zentuit** [playgameservices/play-games-plugin-for-unity#868]
The `Update` method of PlayGamesHelperObject will generate garbage unnecessarily on each frame. In the version I have, the `Update` method looks like this:
````
        public void Update()
        {
            if (sIsDummy || sQueueEmpty)
            {
                return;
            }
            // first copy the shared queue into a local queue
            List<System.Action> q = new List<System.Action>();
            lock (sQueue)
            {
                // transfer the whole queue to our local queue
                q.AddRange(sQueue);
                sQueue.Clear();
                sQueueEmpty = true;
            }

            // execute queued actions (from local queue)
            q.ForEach(a => a.Invoke());
        }
````
There are two places where there are memory allocations on each frame. 

The first is the creation of the local queue `List<System.Action>` instance. To eliminate this, the local queue should be moved out of the method. Then instead of instantiating a new list, use the `Clear()` method in the `Update` method.

The second allocation is the `q.ForEach`. Unity's Mono has a bug that causes memory allocations on `foreach` calls on everything except arrays. Instead, the `ForEach` should be changed to a `for` statement.

These two changes would make the `Update` method look like:
````
       List<System.Action> localQueue = new List<System.Action>(); 

        public void Update()
        {
            if (sIsDummy || sQueueEmpty)
            {
                return;
            }
            // first copy the shared queue into a local queue
            localQueue.Clear();
            lock (sQueue)
            {
                // transfer the whole queue to our local queue
                q.AddRange(sQueue);
                sQueue.Clear();
                sQueueEmpty = true;
            }

            // execute queued actions (from local queue)
            for (int i = 0; i < localQueue.Count; i++) {
                localQueue[i].Invoke();
            }
        }
````
Currently a minimum of 72 B per frame (which is the amount with an empty `q` list) is allocated as garbage. As more `Actions` are added to `sQueue` the amount of garbage allocated will increase (104 B for up to 4 `Actions`, 128 B for 10, 488 B for 100, etc).  While these numbers seem trivial, these numbers are per frame so 60 FPS means 4k of garbage a second.

Making the changes shown will drop the garbage allocation down to 0 B each frame.

**Comment by claywilkinson** [comment]
Nice optimization!  Feel free to submit a pull request to integrate this into the project. Otherwise we'll make a note to update this in the next release.

**Comment by claywilkinson** [comment]
This is included now in the latest plugin, version 0.9.30.

**Comment by Digital-Odyssey** [comment]
I know this is an old thread but I'm currently running version 0.10.14 and the PlayGamesHelperObject is creating around 3.7kb of GC allocations in my mobile game. 

![profiler](https://github.com/playgameservices/play-games-plugin-for-unity/assets/4033028/dab596ac-bb27-432c-bda1-a4590d5b5eba)

On older mobile devices this is giving my game a real performance hit : [

Just wondering if i should stay on this version of the package or upgrade to the latest version hoping it allocates less garbage?
